verifier/src/advisories.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+extern crate alloc;
+
+use crate::{Accessor, SpacedStructName, VerificationOutput, Verifier};
+use alloc::collections::BTreeSet;
+use alloc::string::{String, ToString};
+use core::fmt::{Display, Formatter};
+use serde::Deserialize;
+
+/// The status of a set of advisories
+///
+/// The variants are defined in the schema at
+/// <https://api.portal.trustedservices.intel.com/documentation#pcs-tcb-info-model-v3>
+///
+/// The variant order is important here, the higher the index the better the
+/// status. For example: `UpToDate` is a better status than `SWHardeningNeeded`.
+/// ```
+/// use mc_attestation_verifier::AdvisoryStatus;
+/// assert!(AdvisoryStatus::UpToDate > AdvisoryStatus::SWHardeningNeeded);
+/// ```
+#[derive(Debug, PartialEq, Eq, PartialOrd, Hash, Copy, Clone, Deserialize)]
+pub enum AdvisoryStatus {
+    /// TCB level of SGX platform is revoked. The platform is not trustworthy.
+    Revoked,
+    /// TCB level of SGX platform is outdated and additional configuration of
+    /// SGX platform may be needed.
+    OutOfDateConfigurationNeeded,
+    /// TCB level of SGX platform is outdated.
+    OutOfDate,
+    /// TCB level of the SGX platform is up-to-date but additional configuration
+    /// for the platform and SW Hardening in the attesting SGX enclaves may be
+    /// needed.
+    ConfigurationAndSWHardeningNeeded,
+    /// TCB level of the SGX platform is up-to-date but additional configuration
+    /// of SGX platform may be needed.
+    ConfigurationNeeded,
+    /// TCB level of the SGX platform is up-to-date but due to certain issues
+    /// affecting the platform, additional SW Hardening in the attesting SGX
+    /// enclaves may be needed.
+    SWHardeningNeeded,
+    /// TCB level of the SGX platform is up-to-date.
+    UpToDate,
+}
+
+/// The advisories pertaining to a TCB(Trusted Computing Base).
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct Advisories {
+    ids: BTreeSet<String>,
```

Comment:
This moved to a Set to make the logic simpler, I left the TCB json parsing as a Vec, but once it gets constructed into an advisory here it will become a set, and didn't feel like it helped the TCB stuff much.